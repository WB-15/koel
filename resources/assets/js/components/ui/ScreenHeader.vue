<template>
  <header class="screen-header">
    <div class="thumbnail-wrapper" :class="{ 'non-empty': hasThumbnail }">
      <slot name="thumbnail"></slot>
    </div>

    <div class="heading-wrapper">
      <h1>
        <slot></slot>
      </h1>
      <span class="meta text-secondary">
        <slot name="meta"></slot>
      </span>
    </div>

    <slot name="controls"></slot>
  </header>
</template>

<script lang="ts" setup>
import { toRefs } from 'vue'

const props = withDefaults(defineProps<{ hasThumbnail?: boolean }>(), { hasThumbnail: false })
const { hasThumbnail } = toRefs(props)
</script>

<style lang="scss">
header.screen-header {
  display: flex;
  font-weight: var(--font-weight-thin);
  padding: 1rem 1.8rem;
  border-bottom: 1px solid var(--color-bg-secondary);
  min-height: 96px;
  position: relative;
  align-items: center;
  align-content: stretch;
  line-height: normal;
  gap: 1.5rem;

  .thumbnail-wrapper {
    width: 64px;
    display: none;

    &.non-empty {
      display: block;
    }
  }

  h1 {
    font-size: 2.75rem;
    overflow: hidden;
    white-space: nowrap;
    text-overflow: ellipsis;
  }

  .heading-wrapper {
    overflow: hidden;
    flex: 1;
  }

  .meta {
    display: block;
    font-size: .9rem;
    line-height: 2;
    font-weight: var(--font-weight-light);

    a {
      color: var(--color-text-primary);

      &:hover {
        color: var(--color-highlight);
      }
    }

    > * + * {
      margin-left: .2rem;
      display: inline-block;

      &::before {
        content: 'â€¢';
        margin-right: .2rem;
        color: var(--color-text-secondary);
        font-weight: unset;
      }
    }
  }

  @media (max-width: 768px) {
    min-height: 0;
    flex-direction: column;

    .thumbnail-wrapper {
      display: none;
    }

    h1 {
      font-size: 1.38rem;
    }

    .meta {
      display: none;
    }
  }
}
</style>
