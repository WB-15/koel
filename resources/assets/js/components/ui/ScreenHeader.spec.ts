import { expect, it } from 'vitest'
import ComponentTestCase from '@/__tests__/ComponentTestCase'
import ScreenHeader from './ScreenHeader.vue'

new class extends ComponentTestCase {
  protected test () {
    it('renders', () => {
      expect(this.render(ScreenHeader, {
        slots: {
          default: 'This Header',
          meta: '<p>Some meta</p>',
          controls: '<nav>Some controls</nav>'
        }
      }).html()).toMatchSnapshot()
    })
  }
}
