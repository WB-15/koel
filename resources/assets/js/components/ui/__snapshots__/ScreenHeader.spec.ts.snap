// Vitest Snapshot v1

exports[`renders 1`] = `
<header class="screen-header expanded" data-v-5691beb5="">
  <aside class="thumbnail-wrapper" data-v-5691beb5=""><img src="https://placekitten.com/200/300" data-v-5691beb5-s=""></aside>
  <main data-v-5691beb5="">
    <div class="heading-wrapper" data-v-5691beb5="">
      <h1 class="name" data-v-5691beb5="">This Header</h1><span class="meta text-secondary" data-v-5691beb5=""><p data-v-5691beb5-s="">Some meta</p></span>
    </div>
    <nav data-v-5691beb5-s="">Some controls</nav>
  </main>
</header>
`;
